### DISCUSSION TITLE
I can't understand why my JS script won't do what it's supposed to do.

### DISCUSSION THREAD

**Discussion by sparkls0** [AutomaApp/automa#1353] [Q&A]
Hey there, newbie here..

Okay, so, what I'm trying to do is supposed to be simple. 

I have a variable inside automa which is a chatGPT output with \n\n everywhere and shi. 

I asked this mofoGPT how, using JS and regex, I can turn my text into a coherent one with line jumps. He gave me this script. That I adapted for Automa.

So what it does is retrieve the data from 'gpt' automa variable, puts it in 'inputText' JS const

Line 2 is supposed to make the regex replacements I want it to do

Line 3 is supposed to append a new variable into automa named gpt1, which value would be the new formatted text. 

Thing is, it doesn't work. I know line 1 and 3 are correct, I tested them both. This is how it should be written if I'm not mistaken, with the Automa communication functions. 

But nothing happens when I start the script with line 2 in it, variable gpt1 isn't even created, whatever the output should be. Yet logs give me no errors

I'm confused.. Can someone better than me in code help? I'm just a regular dumbass entrepreneur lmao. 

Thanks in advance boys

<img width="924" alt="CleanShot 2023-09-03 at 22 53 32@2x" src="https://github.com/AutomaApp/automa/assets/48011728/4ff2f983-6c48-4f3f-a323-11e96611b9e6">

**Comment by Delusionnnn** [comment]
Not sure why you dont have an automaNextBlock command at the end as I believe it is needed to finalize as I had no problem with it generating the end variable. I'm not highly skilled but using Regex commands was able to turn a variable that was set to 123456789qwertyz!@#$%^&*() into 123456789qwertyz using the following:

const inputText = automaRefData('variables', 'gpt')
const outputText = inputText.replace(/[^A-Za-z0-9]+/, "");
automaSetVariable('gpt1', outputText)
automaNextBlock()

![image](https://github.com/AutomaApp/automa/assets/141815049/5eb631e2-bd2a-463d-900f-291ed4769c5a)
